### ISSUE TITLE
False positive Generic.ControlStructures.DisallowYodaConditions

### ISSUE THREAD

**Issue by jrfnl** [squizlabs/PHP_CodeSniffer#2962]
The `Generic.ControlStructures.DisallowYodaConditions` reports a `Usage of Yoda conditions is not allowed; switch the expression order` error for the below code, which IMO is incorrect (the error, not the code).

```php
$var = ( (string) function_call( 'text' ) ) === '0';
```

**Comment by umherirrender** [comment]
Another one:

```
					return ( $group && $pos ) ?
						( $pos % $group ) === 0 && ( $pos < 0 === $group < 0 ) :
						!$pos;
```

It seems that a boolean condition with two < 0 is not handled correctly by the sniff.

Work around: Add () around each part of the condition

**Comment by jlherren** [comment]
I'm also experiencing this, here's a very short example:

```
$a = 1 * (1 + 1) !== 0;
```

**Comment by jrfnl** [comment]
Closing as replaced by PHPCSStandards/PHP_CodeSniffer#19
